mino/minows/address_test.go

```diff
@@ -0,0 +1,232 @@
+	reference := mustCreateAddress(AddrHostname, PID1)
+	tests := map[string]struct {
+		self  address
+		other mino.Address
+		out   bool
+	}{
+		"self": {
+			self:  reference,
+			other: reference,
+			out:   true,
+		},
+		"copy": {
+			self:  reference,
+			other: mustCreateAddress(AddrHostname, PID1),
+			out:   true,
+		},
+		"diff location": {
+			self:  reference,
+			other: mustCreateAddress(AddrLocalhost, PID1),
+			out:   false,
+		},
+		"diff identity": {
+			self:  reference,
+			other: mustCreateAddress(AddrHostname, PID2),
+			out:   false,
+		},
+		"nil": {
+			self:  reference,
+			other: nil,
+			out:   false,
+		},
+	}
+	t.Parallel()
+	for name, tt := range tests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			result := tt.self.Equal(tt.other)
+
+			require.Equal(t, tt.out, result)
+		})
+	}
+}
+
+var sharedTests = map[string]struct {
+	addr        address
+	string      string
+	marshalText []byte
+}{
+	"all interface": {
+		addr:        mustCreateAddress(AddrAllInterface, PID1),
+		string:      "/ip4/0.0.0.0/tcp/80/p2p/QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU",
+		marshalText: []byte("/ip4/0.0.0.0/tcp/80/p2p/QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"),
+	},
+	"localhost": {
+		addr:        mustCreateAddress(AddrLocalhost, PID2),
+		string:      "/ip4/127.0.0.1/tcp/80/p2p/QmVt9t5Tk2uEoA4CDbKNCVxqrut8UXmWHXvFZ8wFZ3ghhv",
+		marshalText: []byte("/ip4/127.0.0.1/tcp/80/p2p/QmVt9t5Tk2uEoA4CDbKNCVxqrut8UXmWHXvFZ8wFZ3ghhv"),
+	},
+	"hostname": {
+		addr:        mustCreateAddress(AddrHostname, PID1),
+		string:      "/dns4/example.com/tcp/80/p2p/QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU",
+		marshalText: []byte("/dns4/example.com/tcp/80/p2p/QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"),
+	},
+}
+
+func TestAddress_String(t *testing.T) {
+	t.Parallel()
+	for name, tt := range sharedTests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			result := tt.addr.String()
+
+			require.Equal(t, tt.string, result)
+		})
+	}
+}
+
+func TestAddress_MarshalText(t *testing.T) {
+	t.Parallel()
+	for name, tt := range sharedTests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			result, err := tt.addr.MarshalText()
+
+			require.NoError(t, err)
+			require.Equal(t, tt.marshalText, result)
+		})
+	}
+}
+
+func TestAddressFactory_FromText(t *testing.T) {
+	t.Parallel()
+	factory := addressFactory{}
+	for name, tt := range sharedTests {
+		tt := tt
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			result := factory.FromText(tt.marshalText)
+
+			require.Equal(t, tt.addr, result)
+		})
+	}
+}
+
+func TestAddressFactory_FromText_invalid(t *testing.T) {
+	tests := map[string]struct {
+		in []byte
+	}{
+		"invalid text": {
+			in: []byte("invalid"),
+		},
+		"missing location": {
+			in: []byte("/p2p/QmaD31nEzFGwD8dK96UFWHtTYTqYJgHLMYSFz4W4Hm2WCU"),
+		},
+		"missing identity": {
+			in: []byte(AddrLocalhost),
+		},
+	}
+
+	t.Parallel()
```

Comment:
From what I understand reading https://pkg.go.dev/testing#T.Parallel, this is not needed. Only the one on line 201.